src/Element.js

```diff
@@ -0,0 +1,341 @@
+	this.key = element.key
+ 	this.uuid = element.uuid
+ 	this.type = element.type
+ 	this.props = element.props
+ 	this.children = element.children
+ 	this.ref = null
+ 	this.host = host
+ 	this.xmlns = ''
+ 	this.state = null
+ 	this.owner = owner
+ 	this.parent = parent
+ 	this.active = Constant.active
+}, descriptors)
+
+/**
+ * @type {object}
+ */
+export var prototype = constructor[Constant.prototype]
+
+/**
+ * @constructor
+ * @param {number} uuid
+ * @param {*} type
+ * @param {Array<object>} children
+ * @param {object} props
+ * @param {*} key
+ */
+export var struct = Utility.extend(function node (uuid, type, props, children, key) {
+	this.key = key
+ 	this.uuid = uuid
+ 	this.type = type
+ 	this.props = props
+ 	this.children = children
+}, descriptors)
+
+/**
+ * @param {object} owner
+ * @return {object}
+ */
+export function target (owner) {
+	return new constructor(null, null, create('#root'), owner)
+}
+
+/**
+ * @param {*} element
+ * @return {object}
+ */
+export function module (element) {
+	return from(element && typeof element === 'object' && element.default || element, Constant.key)
+}
+
+/**
+ * @param {*} children
+ * @param {*} target
+ * @param {*} key
+ * @return {object}
+ */
+export function portal (children, target, key) {
+	return create(target, {key: key}, children)
+}
+
+/**
+ * @param {number} type
+ * @param {Array<object>} children
+ * @param {number} key
+ * @return {object}
+ */
+export function iterable (type, children, key) {
+	return new struct(type, '#iterable', Constant.object, children, key)
+}
+
+
+/**
+ * @param {(number|string)} children
+ * @param {*} key
+ * @return {object}
+ */
+export function comment (children, key) {
+	return new struct(Constant.comment, '#comment', Constant.object, children, key)
+}
+
+/**
+ * @param {(number|string)} children
+ * @param {number} index
+ * @return {object}
+ */
+export function text (children, index) {
+	return new struct(Constant.text, '#text', Constant.object, children, Utility.hash(index))
+}
+
+/**
+ * @param {number} index
+ * @return {object}
+ */
+export function empty (index) {
+	return new struct(Constant.empty, '#empty', Constant.object, '', Utility.hash(index))
+}
+
+/**
+ * @param {*} a
+ * @param {(object|*)?} b
+ * @param {...*?}
+ * @return {object}
+ * @public
+ */
+export function create (a, b) {
+	var index = 0
+	var length = arguments.length
+	var i = properties(b) ? 2 : 1
+	var size = length - i
+	var uuid = identity(a)
+	var type = a
+	var props = i === 2 ? b : {}
+	var children = []
+	var key = props.key || Constant.key
+	var element = new struct(uuid, type, props, children, key)
+
+	if (uuid === Constant.component) {
+		if (size) {
+			for (props.children = size === 1 ? arguments[i++] : children = []; i < length; ++i) {
+				children.push(arguments[i])
+			}
+		}
+		if (type[Constant.defaultProps]) {
+			defaults(element, type[Constant.defaultProps])
+		}
+		if ("production" === 'development') {
```

Comment:
?

Reply:
Dummy content that mimics the heuristic of `process.env.NODE_ENV` or `__DEV__` after the fact.